src/client/Client.cc

```diff
@@ -10061,7 +10063,7 @@ int64_t Client::_preadv_pwritev_locked(Fh *fh, const struct iovec *iov,
                resid -= round_size;
                /* iter is self-updating */
         }
-        cl.lock();
+        client_lock.lock();
```

Comment:
Side question...what's the rationale for dropping and reacquiring the lock here in the first place? Isn't this just doing in-memory copies? Does iter.copy() take the client_lock itself or something? Might be worthwhile to just get rid of these calls in a separate patch.

Reply:
> Side question...what's the rationale for dropping and reacquiring the lock here in the first place? Isn't this just doing in-memory copies? Does iter.copy() take the client_lock itself or something? Might be worthwhile to just get rid of these calls in a separate patch.

The `bl` is local parameter in the stack,  and the `iov[]` from caller, no race for both of them, there is need take the `client_lock`, and the copy may take a little long time with lots of data.

> 
> Looking closer, do you even need to reacquire the lock here? It looks like all of the callers eventually just release it without doing anything further. I think it's probably possible to rework this code such that you release it before copying to the iovec, and don't reacquire it.

With this approach, we need to adopt the `client_lock` and then release the lock.

```
std::unique_lock cl{client_lock, std::adopt_lock};
cl.unlock();
// copy data here
cl.release();
```
Is there any perf impovement with this ? Or something else ?

Reply:
Yeah, this PR still one interim, at last in the inode lock patch I will move the `client_lock` or `inode_lock` in the `_preadv_pwritev_locked()` method, then we can just relase the lock before copying.